public/app/features/auth-config/components/ServerDiscoveryModal.tsx

```diff
@@ -0,0 +1,78 @@
+import { JSX } from 'react';
+import { useForm } from 'react-hook-form';
+import { connect, ConnectedProps } from 'react-redux';
+
+import { Button, Input, Field, Modal } from '@grafana/ui';
+
+import { Trans } from '../../../core/internationalization';
+import { ServerDiscoveryFormData } from '../types';
+import { isUrlValid } from '../utils/url';
+
+interface OwnProps {
+  isOpen: boolean | undefined;
+  onClose: () => void;
+  onSuccess: (data: ServerDiscoveryFormData) => void;
+}
+
+export type Props = OwnProps & ConnectedProps<typeof connector>;
+
+const connector = connect(undefined, {});
+
+export const ServerDiscoveryModalUnconnected = ({ isOpen, onClose, onSuccess }: Props): JSX.Element => {
+  const {
+    handleSubmit,
+    register,
+    formState: { errors },
+  } = useForm({
+    mode: 'onBlur',
+    defaultValues: {
+      url: '',
+    },
+  });
+
+  const validateUrl = (value?: string) => {
+    if (value === '') {
+      return 'Please enter the .well-known/openid-configuration endpoint for your IdP';
+    }
+
+    if (!isUrlValid(value)) {
+      return 'Please enter a valid URL';
+    }
+
+    return true;
+  };
+
+  const onSubmit = (data: ServerDiscoveryFormData) => {
+    onSuccess(data);
+  };
+
+  return (
+    <Modal title="OpenID Connect Discovery URL" onDismiss={onClose} onClickBackdrop={onClose} isOpen={isOpen}>
+      <form
+        onSubmit={(e) => {
+          e.stopPropagation();
+          return handleSubmit(onSubmit)(e);
+        }}
```

Comment:
```suggestion
        onSubmit={handleSubmit(onSubmit)}
```
Changing `onSubmit` to this doesn't stop the event propagation?

Reply:
If I don't stop the propagation here then the main form will also be submitted.